pmda/hbond_analysis.py

```diff
@@ -0,0 +1,579 @@
+        u = self._universe()
+        hydrogens_ag = u.select_atoms(hydrogens_sel)
+
+        ag = hydrogens_ag.residues.atoms.select_atoms(
+            "({donors_sel}) and around {d_h_cutoff} {hydrogens_sel}".format(
+                donors_sel=selection,
+                d_h_cutoff=self.d_h_cutoff,
+                hydrogens_sel=hydrogens_sel
+            )
+        )
+        donors_ag = ag[ag.charges < max_charge]
+        donors_list = np.unique(
+            [
+                '(resname {} and name {})'.format(r, p)
+                for r, p in zip(donors_ag.resnames, donors_ag.names)
+            ]
+        )
+
+        return " or ".join(donors_list)
+
+    def guess_acceptors(self, selection='all', max_charge=-0.5):
+        """Guesses which atoms could be considered acceptors in the analysis.
+
+        Parameters
+        ----------
+        selection: str (optional)
+            Selection string for atom group from which acceptors will be
+            identified.
+        max_charge: float (optional)
+            Maximum allowed charge of an acceptor atom.
+
+        Returns
+        -------
+        potential_acceptors: str
+            String containing the :attr:`resname` and :attr:`name` of all
+            atoms that potentially capable of forming hydrogen bonds.
+
+        Notes
+        -----
+        This function makes use of and atomic charges to identify which atoms
+        could be considered acceptor atoms in the hydrogen bond analysis.
+        If an atom has an atomic charge less than :attr:`max_charge` then it
+        is considered capable of participating in hydrogen bonds.
+
+        If :attr:`acceptors_sel` is `None`, this function is called to guess
+        the selection.
+
+        Alternatively, this function may be used to quickly generate a
+        :class:`str` of potential acceptor atoms involved in hydrogen bonding.
+        This :class:`str` may then be modified before being used to set the
+        attribute :attr:`acceptors_sel`.
+        """
+
+        u = self._universe()
+        ag = u.select_atoms(selection)
+        acceptors_ag = ag[ag.charges < max_charge]
+        acceptors_list = np.unique(
+            [
+                '(resname {} and name {})'.format(r, p)
+                for r, p in zip(acceptors_ag.resnames, acceptors_ag.names)
+            ]
+        )
+
+        return " or ".join(acceptors_list)
+
+    def _get_dh_pairs(self, u):
+        """Finds donor-hydrogen pairs.
+
+        Returns
+        -------
+        donors, hydrogens: AtomGroup, AtomGroup
+            AtomGroups corresponding to all donors and all hydrogens.
+            AtomGroups are ordered such that, if zipped, will produce a list
+            of donor-hydrogen pairs.
+        """
+
+        # If donors_sel is not provided, use topology to find d-h pairs
+        if not self.donors_sel:
+
+            if not (hasattr(u, 'bonds') and len(u.bonds) != 0):
+                raise Exception(
```

Comment:
Raise a `ValueError` instead of generic exception.

Is there a kwarg that could be set when instantiating the class? "set HydrogenBondAnalysis.donors_sel" does not really tell me what to do. This should be clearer – maybe add an example to the docs and include a link to the docs??